src/compose/ComposeBox.js

```diff
@@ -153,45 +153,77 @@ const updateTextInput = (textInput, text) => {
   textInput.setNativeProps({ text });
 };
 
-class ComposeBoxInner extends PureComponent<Props, State> {
-  static contextType = ThemeContext;
-  context: ThemeData;
+type LegacySetState<S: { ... }, P> = $PropertyType<React$Component<P, S>, 'setState'>;
+
+/**
+ * Gives setState that takes partial state to merge, like Component.setState.
+ *
+ * See https://reactjs.org/docs/hooks-faq.html#should-i-use-one-or-many-state-variables.
+ *
+ * Unlike Component.setState, and the set-state function from the useState
+ * hook, both of which never change, the value of the returned setState
+ * function changes when props change.
```

Comment:
I think this part is worth avoiding, even though it means giving up the use of `$PropertyType` above and instead spelling out the type. The type doesn't get *that* complicated, and on the other hand the code calling this function gets simpler to think about -- both in itself, and as a change vs. the code before and after.

(It's convenient that we don't actually use the feature of `setState` where its callback can be passed the props as well as the state; if we did, we might not be able to avoid this.)

Here's what the type can be, just copy-pasting from the React libdef and cutting out the features we don't use:
```js
type LegacySetState<S: { ... }> = (partialState: ?$Shape<S> | (S => ?$Shape<S>)) => void;
```

Then `useLegacyState` can drop all mention of the props. (In fact what I did was I edited `useLegacyState` first to drop mention of props, and then used the Flow error message to find the `setState` type definition to copy from.)